### ISSUE TITLE
[🐛] Error with generating a sign/signAsync in NestJS

### ISSUE THREAD

**Issue by gabriiels** [oven-sh/bun#4596]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.2.0-32-generic x86_64 x86_64

### What steps can reproduce the bug?

Using the NestJS framework with Bun, the following error is encountered:

Error with JwtModule, specifically with signAsync/sign.
Generally, the error occurs when encrypting or decrypting.

Works well with Node.

Gives an error with Bun.

Steps to test the version with Bun:

https://github.com/gabriiels/bun-error-jwt-v1

```
bun install

nest build

bun run dist/main.js

```
Steps to test the Node version:

```
bun install

nest start --watch

```
Endpoint:
http://localhost:3000/auth/sign-in

The error is with version

`  "@nestjs/jwt": "^10.1.1",`

With version 9 it works : 

` "@nestjs/jwt": "^9.0.0",`

Error:

```
[Nest] 24837  - 09/08/2023, 1:58:47 PM   ERROR [ExceptionsHandler] Right hand side of instanceof is not an object
TypeError: Right hand side of instanceof is not an object
    at <anonymous> (home/bun/error-fix-jwt/node_modules/jsonwebtoken/sign.js:108:8)
    at new Promise (:1:11)
    at signAsync (home/bun/error-fix-jwt/node_modules/@nestjs/jwt/dist/jwt.service.js:46:142)
    at <anonymous> (home/bun/error-fix-jwt/dist/auth/auth.service.js:26:8)
    at signIn (home/bun/error-fix-jwt/dist/auth/auth.service.js:19:24)
    at <anonymous> (home/bun/error-fix-jwt/dist/auth/auth.controller.js:24:58)
    at signIn (home/bun/error-fix-jwt/dist/auth/auth.controller.js:19:21)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/@nestjs/core/router/router-execution-context.js:37:48)
    at intercept (home/bun/error-fix-jwt/node_modules/@nestjs/core/interceptors/interceptors-consumer.js:10:16)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/@nestjs/core/router/router-execution-context.js:46:51)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/@nestjs/core/router/router-execution-context.js:40:30)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/@nestjs/core/router/router-proxy.js:10:12)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/@nestjs/core/router/router-proxy.js:7:16)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/layer.js:94:21)
    at next (home/bun/error-fix-jwt/node_modules/express/lib/router/route.js:146:11)
    at dispatch (home/bun/error-fix-jwt/node_modules/express/lib/router/route.js:115:21)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/layer.js:94:21)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:283:43)
    at process_params (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:349:6)
    at next (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:280:10)
    at urlencodedParser (home/bun/error-fix-jwt/node_modules/body-parser/lib/types/urlencoded.js:100:6)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/layer.js:94:21)
    at trim_prefix (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:327:41)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:288:13)
    at process_params (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:349:6)
    at next (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:280:10)
    at jsonParser (home/bun/error-fix-jwt/node_modules/body-parser/lib/types/json.js:122:6)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/layer.js:94:21)
    at trim_prefix (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:327:41)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:288:13)
    at process_params (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:349:6)
    at next (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:280:10)
    at expressInit (home/bun/error-fix-jwt/node_modules/express/lib/middleware/init.js:39:9)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/layer.js:94:21)
    at trim_prefix (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:327:41)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:288:13)
    at process_params (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:349:6)
    at next (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:280:10)
    at query (home/bun/error-fix-jwt/node_modules/express/lib/middleware/query.js:44:9)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/layer.js:94:21)
    at trim_prefix (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:327:41)
    at <anonymous> (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:288:13)
    at process_params (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:349:6)
    at next (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:280:10)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/router/index.js:176:21)
    at handle (home/bun/error-fix-jwt/node_modules/express/lib/application.js:180:30)
    at app (home/bun/error-fix-jwt/node_modules/express/lib/express.js:38:29)
    at emit (node:events:122:48)
    at fetch (node:http:377:14)

```

Other error:
https://github.com/oven-sh/bun/issues/1641#issuecomment-1708712998
https://github.com/oven-sh/bun/issues/1641#issuecomment-1708638260

### Additional information

The error is with version

`  "@nestjs/jwt": "^10.1.1",`

With version 9 it works : 

` "@nestjs/jwt": "^9.0.0",`

**Comment by daniklean** [comment]
# Solving Error with generating a sign/signAsync in NestJS #4596

Hello everyone, 

@gabriiels I'm currently looking into your issue, and the problem does not lie with Nest and its dependencies on Bun. The problem arises from a coding practice that can be described as "Hardcore" within your controller. Nest.js does not operate in this manner, neither in Bun nor Node.js.

On the other hand, I have been testing this issue with other developers in the Linux environment, and we encountered the same problem you are currently facing. However, by applying the solution we have developed and as a best practice, the Access Token is now returned. Additionally, a developer with a Mac using an Intel chip has managed to run your project without any modifications. We deduce that there may be a hardware compatibility issue, but we do not want to attribute it to the core of Bun. We are impressed by the results and performance achieved.

Currently, we are ensuring compatibility using virtualization environments, but since Bun is still relatively new, there is still a possibility of errors.

Solution in the controller:

```typescript
import { Body, Controller, Post } from '@nestjs/common';
import { AuthService } from './auth.service';
import { userInterface } from './user.interface';

@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @Post('sign-in')
  async signIn(@Body() { userId, username, password }: userInterface) {
    return await this.authService.signIn(
      { userId, username, password },
      'admin',
    );
  }
}
```

Change in the service:
```typescript
import { Injectable, UnauthorizedException } from '@nestjs/common';
import { JwtService } from '@nestjs/jwt';
import { userInterface } from './user.interface';

@Injectable()
export class AuthService {
  constructor(private jwtService: JwtService) {}

  async signIn({ userId, username, password }: userInterface, pass: string) {
    if (password !== pass) {
      throw new UnauthorizedException();
    }
    const payload = { sub: userId, username };
    return {
      access_token: await this.jwtService.signAsync(payload),
    };
  }
}
```
[Docs for Decorators - Nest ](https://docs.nestjs.com/custom-decorators)  | 
I hope I have helped you.
If you have any doubts or questions, you can reach out to me at @Daniklean on Telegram.

**Comment by gliviu** [comment]
I had the same error while using firebase-admin library.
```
Error sending message: 109 | 
110 |   if (!secretOrPrivateKey && options.algorithm !== 'none') {
111 |     return failure(new Error('secretOrPrivateKey must have a value'));
112 |   }
113 | 
114 |   if (secretOrPrivateKey != null && !(secretOrPrivateKey instanceof KeyObject)) {
                                         ^
TypeError: Right hand side of instanceof is not an object
      at node_modules/jsonwebtoken/sign.js:114:38

```

I can confirm #5940 fixes it.
Just `bun upgrade --canary` and the error is gone.

**Comment by Jarred-Sumner** [comment]
Fixed by @cirospaciari in #5940
